### ISSUE TITLE
Kernel Smoothing クラスの作成

### ISSUE THREAD

**Issue by ae14watanabe** [furukawa-laboratory/somf#41]
**target code 対象となるコード** 明確に分かっているならば書く
flib/libs/models/kernel_smoothing.py（新しく作成する.pyファイル）

**Is your feature request related to a problem? Please describe. この機能追加リクエストは何かの問題に関連していますか？**
SOMのM-stepで用いられるKernel Smoothingであるが，これ単独で回帰手法としてクラスがあれば便利ではないか，という思い付き

**Describe the solution you'd like 望む解決策の説明**
flib/libs/models/kernel_smoothing.pyを作成し，Class KernelSmoothingを定義する．
メソッドとしてはsklearnに準拠して以下のように作成すると良さそう．
- fit(X, y)
- predict(X)
- gradient(X)

ここでgradientも入れておけば，libs/tools/calc_grad_norm_of_ks.pyは不要になる．

メソッド名で参考にしたもの
https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.GaussianProcessRegressor.html
